### PR TITLE
Refactor/react webpack setup

### PR DESCRIPTION
**Pull request by robskrob**
@chadwpry, this PR seeks to address your feedback with regards to:

- package.json
  - use babel-preset-latest vs babel-preset-2015.
  - add eslint-loader.
  - add eslint-plugin-react.
- config/webpack/dev.config.js.
  - rename config file and remove proxy section of webpack dev server.
  - remove hot module replacement plugin setup.
- .gitignore
  - add npm-debug.log to .gitignore
  - remove npm-debug.log
-*.jsx
  - rename *.jsx files to .js because it's not necessary.

If this PR addresses your feedback adequately, please approve the PR. In the meantime, I will move on to your points about cleaning up main.js and using container components in a different PR.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+build/*
```
Comment (chadwpry) [inline .eslintignore]: Nice! I was not familiar w/this file

File: config/webpack/dev.js [hunk 2]
```diff
@@ -6,27 +6,26 @@ module.exports = {
...
   },
   devtool: "source-map",
   entry: {
-    main: "./js/main.jsx"
+    main: "./js/main.js"
   },
   module: {
     loaders: [
-      { test: /.jsx?$/,
+      { test: /\.(js|jsx)$/,
```
Comment (chadwpry) [inline config/webpack/dev.js]: This is not necessary. An easier way to write this is...

/\.jsx?$/

You can write it the same way on the eslint loader

File: config/webpack/dev.js [hunk 3]
```diff
@@ -6,27 +6,26 @@ module.exports = {
...
         loader: "babel-loader",
         exclude: /node_modules/,
         query: { presets: ["es2015", "react"] }
+      },
+      {
+        test: /\.js$/,
```
Comment (chadwpry) [inline config/webpack/dev.js]: Same as abov ^^^

**Comment by chadwpry** [comment]
Other than what was discussed in the comments, all looks good!

File: config/webpack/dev.js (hunk 2 above)
Comment (robskrob) [inline config/webpack/dev.js]: Ah I forgot that `/.jsx?$/` will include files with both, .js and .jsx extensions.

**Pull request merged**